### ISSUE TITLE
MultiSelect showClear Keyboard Trap

### ISSUE THREAD

**Issue by seanh169** [primefaces/primereact#7613]
### Describe the bug

Using _MultiSelect_ component in `10.8.5` there is a keyboard trap on the clear icon when `showClear` is used

https://github.com/user-attachments/assets/ad20ae38-bf59-4204-9dd9-d252d37169bf

### Reproducer

https://stackblitz.com/edit/6h9fpmmz?file=src%2

### System Information

```Shell
System:
    OS: macOS 14.7.1
    CPU: (12) arm64 Apple M2 Pro
    Memory: 137.50 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 16.20.2 - ~/.nvm/versions/node/v16.20.2/bin/node
    npm: 8.19.4 - ~/.nvm/versions/node/v16.20.2/bin/npm
  Browsers:
    Chrome: 131.0.6778.265
    Edge: 131.0.2903.147
    Safari: 18.1.1
```

### Steps to reproduce the behavior

1. Add a few selections to the multiselect
2. Notice clear icon is visible
3. Attempt to keyboard navigate in and out of the clear icon

## This is also easily reproducible on the documentation examples 

https://primereact.org/multiselect/#basic

### Expected behavior

I expect to be able to keyboard navigate in and out of the clear(x) icon within the multiselect to make this accessible.

**Comment by melloware** [comment]
See ticket: https://github.com/primefaces/primereact/issues/7377

@akshayaqburst added this...

**Comment by seanh169** [comment]
> See ticket: [#7377](https://github.com/primefaces/primereact/issues/7377)
> 
> [@akshayaqburst](https://github.com/akshayaqburst) added this...

The issue is if I keyboard navigate into the clear icon(X) WITHOUT pressing enter I cannot navigate away

**Comment by melloware** [comment]
agreed I want to know why @akshayaqburst added this it might need to be removed.  We can check PrimeVue and if PrimeVue doesn't have it we need to remove it.

**Comment by seanh169** [comment]
> [@melloware](https://github.com/melloware) Can I try to fix why we are not able to navigate away from the clear icon? (Instead of reverting those changes!)
> 
> Let me know and I can work on it.

This would be preferred as the clear functionality is helpful to users.

Also: 

Not sure if this helps at all in efforts to fix this, but the tree select component(https://primereact.org/treeselect/#clearicon) has this clear icon and there is no keyboard trap present while keeping the same/similar functionality/appearance.

**Comment by akshayaqburst** [comment]
@melloware @seanh169 Added a commit which fixes the trap issue.
